lib/kb_sourmash/sourmash_utils/SourmashUtils.py

```diff
@@ -299,13 +303,42 @@ def run_sourmash_search(self, params):
         signature_file = self._build_signatures(assembly_file, params['scaled'], track_abundance)
 
         # run search
+        outpath = os.path.join(self.tmp, 'sourmash_temp_out')
         search_command = [self.SOURMASH_SEARCH, signature_file, search_db,
-                          '-n', str(20), containment_flag]
+                          '-n', str(max_returned), containment_flag, '-o', outpath]
         self._run_command(' '.join(search_command))
 
+        id_to_similarity, ttlcount = self._parse_search_results(outpath, max_returned)
+        print(id_to_similarity)
+        print(len(id_to_similarity))
+        print(ttlcount)
+        # TODO: handle special case of KBase IDs
+
         results = {'report_name': '', 'report_ref': ''}
         return results
 
+    def _parse_search_results(self, results_path, maxcount):
+        lines = self._count_lines(results_path) - 1  # -1 for header
+        id_to_similarity = {}
+
+        with open(results_path, 'rb') as fh:
+            csvfile = csv.DictReader(fh)
+            count = 0
+            for line in csvfile:
+                if count >= maxcount:
```

Comment:
Exit condition doesn't seem to take the header into account. maxcount should be the number of returned hits

Reply:
DictReader automatically takes the header into account; that's how the code is able to access the line fields by name rather than by position in the next line.